### ISSUE TITLE
Use some kind of BigInt for stats?

### ISSUE THREAD

**Issue by hbina** [BurntSushi/xsv#350]
I have data set that involves millions of rows of timestamps and I want to perform some basic stats on it.

**Comment by janxkoci** [comment]
I have data with billions of rows and I didn't run into problems - can you provide more context about what exactly do you need?

**Comment by hbina** [comment]
Its not the number of rows itself that is the problem. Consider what happens if you have billions of rows of UNIX timestamps, the sum, average, median, etc of that column is entirely wrong because it overflowed.

**Comment by janxkoci** [comment]
Okay, most other tools, like GNU awk or [miller](https://miller.readthedocs.io/en/6.13.0/reference-main-arithmetic/) use double to avoid overflowing, would that help?

**Comment by janxkoci** [comment]
I was gonna point out the fork qsv, but it seems to handle the problem [the same way](https://github.com/dathere/qsv/wiki/Supplemental#stats-command-output-explanation).

But then it got me thinking - what possible use can have the _sum_ of timestamps? Arguably, this value can be safely ignored for most purposes, since it's almost surely meaningless. Other values like median or average are of course meaningful, but should be generally smaller (on the order of one timestamp).

I mean, I work in genomics, and even though I could try to calculate the sum of all positions in a genome, this value has no meaning (unlike range or mean).

**Comment by hbina** [comment]
I guess the usage is pretty niche, but I thought it would be nice to support anyways. I don't remember why I needed it at some point.

**Comment by janxkoci** [comment]
Fair enough. For what it's worth, I've done the above tests also with miller and it was the only tool that didn't overflow. If you ever need it again, give miller a spin, it's pretty cool - I use it pretty much every day 😉
